src/overlays/actors/ovl_Obj_Boat/z_obj_boat.c

```diff
@@ -28,31 +32,164 @@ const ActorInit Obj_Boat_InitVars = {
     (ActorFunc)ObjBoat_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B9B680[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 1000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80B9B680[];
-
-extern UNK_TYPE D_06007630;
-extern UNK_TYPE D_06009A88;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/func_80B9AF50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/ObjBoat_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/ObjBoat_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/func_80B9B124.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/ObjBoat_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/func_80B9B428.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boat/ObjBoat_Draw.s")
+s16 func_80B9AF50(ObjBoat* this, Vec3f* arg0) {
+    s16 yaw;
+    Vec3s* temp = &this->unk164[(s32)this->unk15C];
```

Comment:
Is the `s32` cast needed?

Reply:
doesn't match otherwise, i tried converting u8 unk15C (now posX) to an s32 but it doesn't match either.